src/meta/client/MetaClient.cpp

```diff
@@ -157,6 +158,36 @@ bool MetaClient::loadUsersAndRoles() {
     return true;
 }
 
+bool MetaClient::loadSessions() {
+    auto ret = listSessions().get();
+    if (!ret.ok()) {
+        LOG(ERROR) << "List sesssions failed, status:" << ret.status();
+        return false;
+    }
+    decltype(sessionsCache_)       SessionMap;
+    for (auto& sess : ret.value()) {
+        auto& addr = sess.get_addr();
+        auto sessionId = sess.get_session_id();
+        auto startTime = sess.get_start_time();
+        auto updateTime = sess.get_update_time();
+        SessionMap[addr][sessionId] = std::make_pair(startTime, updateTime);
+    }
+    {
+        folly::RWSpinLock::WriteHolder holder(localCacheLock_);
+        sessionsCache_ = std::move(SessionMap);
+    }
+    return true;
+}
+
+bool MetaClient::uploadSession() {
+    auto ret = updateSession(sessions_).get();
```

Comment:
maybe there is no need pass the class member sessions_ by value, the member function updateSession can use it directly

Reply:
Good point